Apps/Src/UpdateVelocity.c

```diff
@@ -22,13 +22,44 @@ void Task_UpdateVelocity(void* p_arg){
         //Read global pedal percentage
         uint8_t pedalPercentage = car_state->AccelPedalPercent;
 
-        if(car_state->CruiseControlEnable && car_state -> CruiseControlSet){
+        if(car_state->CREnable == CRUISE && car_state -> CruiseControlSet){
             car_state->DesiredVelocity = car_state->CruiseControlVelocity;
             car_state->DesiredMotorCurrent = 1.0f;
-        }else{
+
+            car_state->RegenButtonMode = OFF;
+        }
+        else if(car_state->CREnable == REGEN){
+            switch(car_state->RegenButtonMode){
+                case OFF:
+                    car_state->CREnable = OFF;
+                    break;
+                case RATE1:
+                    car_state->CREnable = REGEN;
```

Comment:
We only get to this point if CREnable is set to REGEN, so there's not reason to set it again.